### ISSUE TITLE
spack fetch --dependencies attempts to download 'buildable: False'

### ISSUE THREAD

**Issue by balay** [LLNL/spack#6123]
`spack fetch --dependencies`  attempts to download 'buildable: False' packages - and then aborts when things are not found.

Ideally it should not attempt to download such stuff.

or provide an option to continue with downloading other packages [on error with current package]

**Comment by balay** [comment]
package.py
  ```
hdf5:
    modules:
      hdf5@1.10.0.3%intel@18.0.0.128+hl+fortran arch=cray-CNL-mic_knl: cray-hdf5-parallel/1.10.0.3
    buildable: False
```

```
balay@thetalogin5:~/spack> spack fetch --dependencies xsdk
==> Using cached archive: /gpfs/mira-home/balay/spack/var/spack/cache/xsdk/xsdk-0.3.0-rc1.tar.gz
==> Using cached archive: /gpfs/mira-home/balay/spack/var/spack/cache/alquimia/alquimia-xsdk-0.3.0.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> Using cached archive: /gpfs/mira-home/balay/spack/var/spack/cache/cmake/cmake-3.5.2.tar.gz
==> Warning: There is no checksum on file to fetch hdf5@1.10.0.3 safely.
==>   Fetch anyway? [y/N] y
==> Fetching with no checksum.
  Add a checksum or use --no-checksum to skip this check.
==> Fetching https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.0.3/src/hdf5-1.10.0.3.tar.gz

curl: (22) The requested URL returned error: 404 Not Found
==> Fetching from https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.0.3/src/hdf5-1.10.0.3.tar.gz failed.
==> Can not find version 1.10.0.3 in url_list
==> Error: All fetchers failed for hdf5-1.10.0.3-e3fhnr5otsva35cmpfa4w6dcjk6o6w4h
balay@thetalogin5:~/spack> 

```

**Comment by scheibelp** [comment]
This should be resolved by #6124. Feel free to reopen if the issue comes up for you again.
